### ISSUE TITLE
Open the Shell on the Right Hand Side

### ISSUE THREAD

**Issue by aivarannamaa** [thonny/thonny#158]
**[Original report](https://bitbucket.org/bitbucket-issue-migration\thonny-issues.zip/issue/158) by Grant Carthew (Bitbucket: [grantcarthew](https://bitbucket.org/grantcarthew)
 GitHub: [grantcarthew](https://github.com/grantcarthew)
).**

----------------------------------------

With wide screen monitors it would be great to be able to open the shell on the right, or left.

**Comment by lurch** [comment]
It would be great if there was a way within Thonny itself to change where the windows got docked to, without having to quit Thonny, edit `configuration.ini` and relaunch Thonny :slightly_smiling_face:

**Comment by lurch** [comment]
It's a little bit clunky, as it requires a Thonny restart rather than being done in "real time" (and it's a bit of a copy'n'paste bodge-job), but here's a prototype GUI for changing the dock-window positions :laughing:

```python3
from tkinter import ttk
from thonny.config_ui import ConfigurationPage
from thonny import get_workbench


class DockConfigurationPage(ConfigurationPage):
    def __init__(self, master):
        super().__init__(master)

        row = 0
        for view_id in sorted(get_workbench()._view_records):
            textlabel = get_workbench()._view_records[view_id]["label"]
            label = ttk.Label(self, text=textlabel, anchor="w")
            label.grid(row=row, column=0, sticky="w", pady=5)
            self.add_checkbox(
                "view." + view_id + ".visible",
                _("visible"),
                row=row,
                column=1,
                padx=5
            )
            self.add_combobox(
                "view." + view_id + ".location",
                ["nw", "w", "sw", "s", "ne", "e", "se"], # workbench.py:767
                width=3,
                row=row,
                column=2,
                padx=5,
            )
            row += 1

        reopen_label = ttk.Label(
            self,
            text=_("NB! Restart Thonny after changing these options!"),
            font="BoldTkDefaultFont",
            anchor="w",
        )
        reopen_label.grid(row=100, column=0, pady=20, columnspan=3)

        self.columnconfigure(2, weight=1)


def load_plugin():
    get_workbench().add_configuration_page("dock", _("Docks"), DockConfigurationPage, 50)
```
 
Save it as `dock_config_page.py` inside your `thonny/lib/python3.7/site-packages/thonny/plugins/` directory.
It then looks like:
![image](https://user-images.githubusercontent.com/476186/82806970-cd8f2600-9e7e-11ea-9409-bec65ece7960.png)

**Comment by aivarannamaa** [comment]
Thank you for the suggestion, but I hope to implement tab reordering by drag-and-drop or via left-click menu on tabs.

**Comment by lurch** [comment]
> I hope to implement tab repositioning by drag-and-drop or via left-click menu on tabs.

Yes, that would be much better :grinning:

**Comment by aivarannamaa** [comment]
Implemented under #2737
